x-pack/solutions/search/plugins/search_indices/public/components/shared/sample_data_action_button.tsx

```diff
@@ -30,12 +30,17 @@ interface SampleDataActionButtonProps {
 }
 
 export const SampleDataActionButton: React.FC<SampleDataActionButtonProps> = ({
-  isLoading,
   onIngestSampleData,
   hasRequiredLicense,
 }) => {
   const { application, http, share, uiSettings } = useKibana().services;
-  const { isInstalled, indexName, dashboardId, isLoading: isStatusLoading } = useSampleDataStatus();
+  const {
+    isInstalled,
+    indexName,
+    dashboardId,
+    isLoading: isStatusLoading,
+    isInstalling,
+  } = useSampleDataStatus();
```

Comment:
Since the hook creates the success/failure toast, will the notification of completion only show up when they are on the sample data page? If we expect users to navigate away from the page, is there any existing way to register the react query call in a higher level component like the nav?